### DISCUSSION TITLE
When does pnpm create `_` type links vs direct links? And how can I control it? It's breaking our mono repo setup.

### DISCUSSION THREAD

**Discussion by sean256** [pnpm/pnpm#4508] [Q&A]
We have a mono-repo where 5 packages depend on `@someScope/react@0.6.4`. For some reason which I cannot figure out, `pnpm` creates a _different_ link to a different instance for just ONE of the 5. One of them has `_react-is@17.0.2` in it's name.

I don't want to get into the details on why this is causing us trouble, as I don't want to spin this topic in the wrong direction. But I really want to know the logic here so I can work on syncing this up.

I will say that when I go in and manually delete and re-link the problem child, all of our issues go away.

Also, all of our packages have the same version of `react-is@17.0.2` listed, and so does `@someScope/react@0.6.4`

```
// the oddball link
../../../../node_modules/.pnpm/@someScope+react@0.6.4_react-is@17.0.2/node_modules/@someScope/react
// the normal link
../../../../node_modules/.pnpm/@someScope+react@0.6.4_4db1f9ae395425ec4d56821dc57dd590/node_modules/@someScope/react
```

If this has been answered before I apologies, I honestly couldn't figure out the right search terms for this.

**Comment by zkochan** [comment]
Looks like either `@someScope/react` or one of its dependencies has peer dependencies. At some place in your dependency graph, the peer dependencies are resolved, while at some other place the peer dependencies are not found. You can look into the lockfile, search for the entry of `_4db1f9ae395425ec4d56821dc57dd590` and `_react-is@17.0.2` and see how their dependencies differ.

**Reply by sean256** [reply]
If I could hug you through the internet I would. That was so helpful and with it I solved the issue.

Is there a flag or something that can be used to highlight these issues? Sort of link Lerna's host strict mode or something?

**Reply by zkochan** [reply]
Not exactly like that, but setting [strict-peer-dependencies](https://pnpm.io/npmrc#strict-peer-dependencies) to `true` might help.
